rust/shalom_dart_codegen/templates/operation.dart.jinja

```diff
@@ -143,4 +175,24 @@ class {{ root_type_name }}{
     }
 {% endfor %}
 
-// ------------ END OBJECT DEFINITIONS -------------
+
+
+// ------------ Enum DEFINITIONS -------------
+{% for full_name, selection in operation.context.type_defs | items  if selection.kind == "Enum" %}
+     enum {{ full_name }}  {
+         {% for name, value in selection.concrete_type.values | items %}  
+              {{ value.value }},
+         {% endfor %};
+         
+         static {{ full_name }} fromString(String value) {
+            return {{ full_name }}.values.firstWhere(
+                (e) => e.name.toUpperCase() == value.toUpperCase(), 
+                orElse: () => throw ArgumentError("Unknown {{ full_name }} $value"),
+            );
+         }
```

Comment:
since this could be statically generated I won't rely on dart features here.
this is how I would implement this.
i.e
```dart
Status fromString(String name) {
  switch (name) {
    case 'pending':
      return Status.pending;
    case 'active':
      return Status.active;
    case 'inactive':
      return Status.inactive;
    case 'deleted':
      return Status.deleted;
    default:
      throw ArgumentError.value(name, 'name', 'No Status enum member with this name');
  }
}

void main() {
  print(Status.fromString('active'));
  try {
    Status.fromString('unknown');
  } catch (e) {
    print(e); // Output: Invalid argument(s): 'unknown': No Status enum member with this name
  }
}
```